src/components/ArticlePreview.jsx

```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+import { Link } from 'react-router';
+
+export default class ArticlePreview extends React.Component {
+
+  loadPost(e) {
```

Comment:
I don't think this is needed right? Navigating via <Link> will cause a re-render in react router, which will cause a new controller to be mounted which will automatically fetch the new data.